File: web-face/src/app/core/services/alert.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { Alert } from '../models/alert.model';
import { AlertStatus } from '../models/enums';
import { CrudService } from './crud.service';
import { buildHttpParams } from './http-params.util';

export interface AlertSearchParams {
  deviceId?: string;
  status?: AlertStatus;
}

@Injectable({ providedIn: 'root' })
export class AlertService extends CrudService<Alert> {
  constructor() {
    super('alerts');
  }

  /** Provide at least one of deviceId or status, matching the backend's /search contract. */
  search(params: AlertSearchParams): Observable<Alert[]> {
    return this.http.get<Alert[]>(`${this.resourceUrl}/search`, { params: buildHttpParams(params) });
  }

  countByStatus(status: AlertStatus): Observable<number> {
    return this.http.get<number>(`${this.resourceUrl}/count`, { params: buildHttpParams({ status }) });
  }
}

File: core/src/main/java/com/csl/lasform/service/AlertService.java
package com.csl.lasform.service;

import java.util.List;

import com.csl.lasform.model.entity.Alert;
import com.csl.lasform.model.entity.enums.AlertStatus;

public interface AlertService extends CrudService<Alert, String> {

    List<Alert> findByDeviceId(String deviceId);

    List<Alert> findByStatus(AlertStatus status);

    List<Alert> findByDeviceIdAndStatus(String deviceId, AlertStatus status);

    long countByStatus(AlertStatus status);
}
